experiment/simulation/js/main.js

```diff
@@ -1 +1,234 @@
+		];
+
+		ball = [up + build_wid + 300, buildY + 100];
+		
+		dirn = 0;
+		collis = 0;
+		scale = 4;
+
+		tmHandle = window.setTimeout(draw, 1000 / fps);
+	}
+
+	const slider_wid = document.getElementById("lineWidth");
+	const output_wid = document.getElementById("demo_width");
+	output_wid.innerHTML = slider_wid.value; // Display the default slider value
+
+	slider_wid.oninput = function() {
+		output_wid.innerHTML = this.value;
+		lineWidth = 0.5 * Number(document.getElementById("lineWidth").value);
+		stiff = 0.15 * Number(document.getElementById("lineWidth").value);
+		fps = 3 + Number(document.getElementById("lineWidth").value); 
+		restart();
+	};
+
+	const slider_mot = document.getElementById("motion");
+	const output_mot = document.getElementById("demo_motion");
+	output_mot.innerHTML = slider_mot.value; // Display the default slider value
+
+	slider_mot.oninput = function() {
+		output_mot.innerHTML = this.value;
+		vibe = Number(document.getElementById("motion").value);
+		restart();
+	};
+
+	const slider_mas = document.getElementById("mass");
+	const output_mas = document.getElementById("demo_mass");
+	output_mas.innerHTML = slider_mas.value; // Display the default slider value
+
+	slider_mas.oninput = function() {
+		output_mas.innerHTML = this.value;
+		text = Number(document.getElementById("mass").value);
+		mass_stiff = (0.2 * Number(document.getElementById("mass").value)) / 1000;
+		restart();
+	};	
+
+	function drawBldg(ctx, v)
+	{
+		ctx.beginPath();
+		ctx.moveTo(v[0][0], v[0][1]);
+		for(let ind = 1; ind < v.length; ++ind)
+		{
+			ctx.lineTo(v[ind][0], v[ind][1]);
+		}
+		ctx.closePath();
+		ctx.fill();
+		ctx.stroke();
+	}
+
+	function drawGround(ctx, ground)
+	{
+		ctx.save();
+		ctx.fillStyle = "pink";
+		ctx.lineWidth = 1.5;
+		ctx.beginPath();
+		ctx.moveTo(ground[0][0], ground[0][1]);
+
+		for(let i = 0; i < ground.length; ++i)
+		{
+			const next = (i + 1) % ground.length;
+			ctx.lineTo(ground[next][0], ground[next][1]);
+		}
+
+		ctx.closePath();
+		ctx.fill();
+		ctx.stroke();
+		ctx.restore();
+	}
+
+	let vibe = 75;
+
+	const canvas = document.getElementById("main");
+	canvas.width = 1200;
+	canvas.height = 600;
+	canvas.style = "border:3px solid";
+	const ctx = canvas.getContext("2d");
+
+	const border = "black";
+	const fill = "#55fff7";
+	let fps = 10;
+	let lineWidth = 3.5;
+
+	let dirn = 0;
+	let scale = 4;
+	let stiff = 1.05;
+	let mass_stiff = 0.6;
+	let text = 3000;
+
+	const up = 400;
+	const buildY = 100;
+	const build_height = 375;
+	const build_wid = 140;
+	const build_3dwid = 35;
+
+	let bldg = [
+		[up, buildY], 
+		[up + build_wid, buildY], 
+		[up + build_wid - build_3dwid, buildY + build_3dwid], 
+		[up - build_3dwid, buildY + build_3dwid], 
+		[up + build_wid, buildY + build_height], 
+		[up + build_wid - build_3dwid, buildY + build_3dwid + build_height], 
+		[up - build_3dwid, buildY + build_3dwid + build_height],
+	];
+
+	let ball = [up + build_wid + 300, buildY + 100];
+
```

Comment:
Can the above ~10 lines be moved to an initializer function which can be called from here as well as from the restart function?

Reply:
Added the initializer function